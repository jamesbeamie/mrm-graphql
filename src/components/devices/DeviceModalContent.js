import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Input, SelectInput, ActionButtons } from '../commons';

const deleteMessage = formData =>
  (
    <p id="confirm-msg">
      Are you sure you want to delete &quot;{`${formData.name}: ${formData.deviceType}`}
      &quot;? <br />
      This cannot be undone
    </p>
  );

const innerForm = (formData, methods, data) =>
  (
    <Fragment>
      <Input
        id="device-name"
        name="device-name"
        placeholder="Enter Device Name"
        labelName="Device Name"
        labelClass="device-input"
        value={formData.name}
        onChange={methods.handleNameChange}
      />
      <SelectInput
        labelText="Device Type"
        name="device-type"
        id="add-devices-type"
        isValue
        value={formData.deviceType || 'Select Device'}
        options={data.deviceTypes}
        onChange={methods.handleTypeChange}
        wrapperClassName="device-input"
        placeholder="Select Device"
        selectInputClassName="dynamic-input-field default-select"
      />
      <SelectInput
        labelText="Room"
        name="room"
        id="add-devices-rooms"
        value={formData.roomId || 'Select Room'}
        options={data.rooms}
        onChange={methods.handleRoomChange}
        wrapperClassName="device-input"
        placeholder="Select Room"
        selectInputClassName="dynamic-input-field default-select"
      />
    </Fragment>
  );

const DeviceModalContent = (props) => {
  const { formData, methods } = props;
  const {
    modalContent, rooms, deviceTypes, fetching,
  } = props;
  return (
    <form className="amenity-form device-type">
      <div>
        {modalContent !== 'delete' ? innerForm(formData, methods, { deviceTypes, rooms })
          : deleteMessage(formData)}
        <div className="loading-btn-div">
          <ActionButtons
            withCancel
            isLoading={fetching}
            onClickCancel={methods.closeModal}
            actionButtonText={`${modalContent ? ((modalContent !== 'delete') ?
              'SAVE CHANGES' : 'DELETE') : 'SUBMIT'} `}
            onClickSubmit={methods.submit}
          />
        </div>
      </div>
    </form>);
};

DeviceModalContent.propTypes = {
  formData: PropTypes.shape({
    name: PropTypes.string,
    roomId: PropTypes.string,
    deviceType: PropTypes.string,
  }).isRequired,
  methods: PropTypes.shape({
    handleNameChange: PropTypes.func,
    handleRoomChange: PropTypes.func,
    handleTypeChange: PropTypes.func,
    closeModal: PropTypes.func,
    submit: PropTypes.func,
  }).isRequired,
  fetching: PropTypes.bool,
  modalContent: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  rooms: PropTypes.instanceOf(Array).isRequired,
  deviceTypes: PropTypes.instanceOf(Array).isRequired,
};

DeviceModalContent.defaultProps = {
  fetching: false,
  modalContent: undefined,
};

export default DeviceModalContent;
