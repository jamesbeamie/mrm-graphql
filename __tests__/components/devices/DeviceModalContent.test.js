import React from 'react';
import { shallow } from 'enzyme';
import DeviceModalContent from '../../../src/components/devices/DeviceModalContent';


jest.mock('../../../src/components/helpers/mutationHelpers/devices');

describe('Device modal content tests', () => {
  let wrapper;
  const props = {
    formData: {
      name: '',
      deviceType: '',
      roomId: '',
    },
    methods: {
      handleNameChange: jest.fn(),
      handleTypeChange: jest.fn(),
      handleRoomChange: jest.fn(),
      closeModal: jest.fn(),
      submit: jest.fn(),
    },
    rooms: [],
    deviceTypes: [],
    fetching: false,
  };
  beforeAll(() => {
    wrapper = shallow(<DeviceModalContent
      {...props}
    />);
  });

  it('Should render for "add" correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render for "delete" correctly', () => {
    wrapper.setProps({
      modalContent: 'delete',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render for "edit" correctly', () => {
    wrapper.setProps({
      modalContent: 'edit',
      formData: {
        name: 'foo',
        deviceType: 'bar',
        roomId: '32',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
