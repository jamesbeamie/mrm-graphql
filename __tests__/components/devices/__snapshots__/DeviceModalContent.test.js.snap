// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Device modal content tests Should render for "add" correctly 1`] = `
<form
  className="amenity-form device-type"
>
  <div>
    <Input
      controlsClass=""
      id="device-name"
      inputClass="mrm-input default-input"
      labelClass="device-input"
      labelName="Device Name"
      name="device-name"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      placeholder="Enter Device Name"
      type="text"
      value=""
    />
    <SelectInput
      id="add-devices-type"
      isValue={true}
      labelText="Device Type"
      name="device-type"
      onChange={[MockFunction]}
      onClick={[Function]}
      options={Array []}
      placeholder="Select Device"
      placeholderValue=""
      required={false}
      selectInputClassName="dynamic-input-field default-select"
      selected={-1}
      value="Select Device"
      wrapperClassName="device-input"
    />
    <SelectInput
      id="add-devices-rooms"
      isValue={false}
      labelText="Room"
      name="room"
      onChange={[MockFunction]}
      onClick={[Function]}
      options={Array []}
      placeholder="Select Room"
      placeholderValue=""
      required={false}
      selectInputClassName="dynamic-input-field default-select"
      selected={-1}
      value="Select Room"
      wrapperClassName="device-input"
    />
    <div
      className="loading-btn-div"
    >
      <ActionButtons
        actionButtonText="SUBMIT "
        cancelButtonText="CANCEL"
        isLoading={false}
        onClickCancel={[MockFunction]}
        onClickSubmit={[MockFunction]}
        showActionButton={true}
        withCancel={true}
      />
    </div>
  </div>
</form>
`;

exports[`Device modal content tests Should render for "delete" correctly 1`] = `
<form
  className="amenity-form device-type"
>
  <div>
    <p
      id="confirm-msg"
    >
      Are you sure you want to delete "
      : 
      "? 
      <br />
      This cannot be undone
    </p>
    <div
      className="loading-btn-div"
    >
      <ActionButtons
        actionButtonText="DELETE "
        cancelButtonText="CANCEL"
        isLoading={false}
        onClickCancel={[MockFunction]}
        onClickSubmit={[MockFunction]}
        showActionButton={true}
        withCancel={true}
      />
    </div>
  </div>
</form>
`;

exports[`Device modal content tests Should render for "edit" correctly 1`] = `
<form
  className="amenity-form device-type"
>
  <div>
    <Input
      controlsClass=""
      id="device-name"
      inputClass="mrm-input default-input"
      labelClass="device-input"
      labelName="Device Name"
      name="device-name"
      onBlur={[Function]}
      onChange={[MockFunction]}
      onFocus={[Function]}
      placeholder="Enter Device Name"
      type="text"
      value="foo"
    />
    <SelectInput
      id="add-devices-type"
      isValue={true}
      labelText="Device Type"
      name="device-type"
      onChange={[MockFunction]}
      onClick={[Function]}
      options={Array []}
      placeholder="Select Device"
      placeholderValue=""
      required={false}
      selectInputClassName="dynamic-input-field default-select"
      selected={-1}
      value="bar"
      wrapperClassName="device-input"
    />
    <SelectInput
      id="add-devices-rooms"
      isValue={false}
      labelText="Room"
      name="room"
      onChange={[MockFunction]}
      onClick={[Function]}
      options={Array []}
      placeholder="Select Room"
      placeholderValue=""
      required={false}
      selectInputClassName="dynamic-input-field default-select"
      selected={-1}
      value="32"
      wrapperClassName="device-input"
    />
    <div
      className="loading-btn-div"
    >
      <ActionButtons
        actionButtonText="SAVE CHANGES "
        cancelButtonText="CANCEL"
        isLoading={false}
        onClickCancel={[MockFunction]}
        onClickSubmit={[MockFunction]}
        showActionButton={true}
        withCancel={true}
      />
    </div>
  </div>
</form>
`;
